meteor/server/api/ingest/rundownInput.ts

```diff
@@ -132,8 +132,12 @@ export enum RundownSyncFunctionPriority {
 export function rundownPlaylistSyncFunction<T extends Function>(
 	rundownPlaylistId: RundownPlaylistId,
 	priority: RundownSyncFunctionPriority,
+	context: string,
 	fcn: T
 ): ReturnType<T> {
+	if (!fcn.name) {
+		Object.defineProperty(fcn, 'name', { value: context })
+	}
 	return syncFunction(fcn, `ingest_rundown_${rundownPlaylistId}`, undefined, priority)()
```

Comment:
If going this way, I think we should add the context argument into the syncFunction as well. Thereby removing the need to use `getFunctionName` altogether (I've seen reports on that it performs poorly, performance-wise)

Reply:
I agree. I'll take a look at extending this PR to all sync functions. Should just be _turning a handle_ across the code base.

Reply:
Personally I am leaning towards the context being a required parameter in all of the sync functions, and to always use that instead of `fcn.name`.
It does make the syncFunction api a little messier/harder to use, but means we can write some proper contexts of where/what the function is, which will aid in tracing and debugging